src/pkg/podman/podman.go

```diff
@@ -332,3 +334,67 @@ func SystemMigrate(ociRuntimeRequired string) error {
 
 	return nil
 }
+
+type internalError struct {
+	Err error
+}
+
+func (e *internalError) Error() string {
+	if e.Err == nil {
+		return ""
+	}
+	return e.Err.Error()
+}
+
+func (e *internalError) Wrap(format string, a ...interface{}) {
+	errMsg := fmt.Sprintf(format, a...)
+	if e.Err == nil {
+		e.Err = errors.New(errMsg)
+	} else {
+		e.Err = fmt.Errorf("%v: %w", errMsg, e.Err)
+	}
+}
+
+// Is wraps around errors.Is() and add support for string comparison of errors
+func (e internalError) Is(err error) bool {
```

Comment:
Is this really the idiomatic way of unwrapping a chain of errors? The [convention](https://blog.golang.org/go1.13-errors) seems to be to offer an `Unwrap` method.

Reply:
Reading the rest of the code and thinking about it more, I am wondering if there are two different approaches to implementing `internalError`.

One is to have it wrap an inner `error`, that could be another `internalError` wrapping yet another `error`, and so on. In that case `internalError` would implement an `Unwrap` method which would return the inner `error`, and `errors.Is` would either match the innermost `error` or a suffix of the chain of errors.

Second is to keep a slice of strings, each representing a layer in Podman's call chain that adds its own context to the error string that gets written to `stderr`. In that case `internalError` would implement an `Unwrap` method that would create another `internalError` by dropping the first element from the slice, and an `Is` method that can match any single member of the slice (ie., any specific layer in Podman) like `grep`.

Notice that there's a subtle difference in the behaviour of `errors.Is` between the two options. Given that parsing `stderr` of any command is a very fragile operation, I think, the second option is more robust. We can't rely on a suffix of the chain of errors remaining unchanged, and it looks like ultimately that's how the code is using `internalError`.

I think what's bothering me about the current implementation is that we are first parsing Podman's `stderr` by splitting the different layers, constructing the chain of errors and then parsing (or splitting) them again in `Is`. We should do it only once.

Reply:
I updated the implementation to match the second option of the two you proposed.